homeassistant/components/renault/services.py

```diff
@@ -0,0 +1,194 @@
+
+from .const import DOMAIN
+from .renault_hub import RenaultHub
+from .renault_vehicle import RenaultVehicleProxy
+
+LOGGER = logging.getLogger(__name__)
+
+ATTR_CHARGE_MODE = "charge_mode"
+ATTR_SCHEDULES = "schedules"
+ATTR_TEMPERATURE = "temperature"
+ATTR_VIN = "vin"
+ATTR_WHEN = "when"
+
+REGEX_VIN = "(?i)^VF1[\\w]{14}$"
+
+SERVICE_AC_CANCEL_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_VIN): cv.matches_regex(REGEX_VIN),
+    }
+)
+SERVICE_AC_START_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_VIN): cv.matches_regex(REGEX_VIN),
+        vol.Required(ATTR_TEMPERATURE): cv.positive_float,
+        vol.Optional(ATTR_WHEN): cv.datetime,
+    }
+)
+SERVICE_CHARGE_SET_MODE_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_VIN): cv.matches_regex(REGEX_VIN),
+        vol.Required(ATTR_CHARGE_MODE): vol.In(
+            ["always", "always_charging", "schedule_mode"]
+        ),
+    }
+)
+SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA = vol.Schema(
+    {
+        vol.Required("startTime"): cv.string,
+        vol.Required("duration"): cv.positive_int,
+    }
+)
+SERVICE_CHARGE_SET_SCHEDULE_SCHEMA = vol.Schema(
+    {
+        vol.Required("id"): cv.positive_int,
+        vol.Optional("activated"): cv.boolean,
+        vol.Optional("monday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("tuesday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("wednesday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("thursday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("friday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("saturday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("sunday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+    }
+)
+SERVICE_CHARGE_SET_SCHEDULES_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_VIN): cv.matches_regex(REGEX_VIN),
+        vol.Required(ATTR_SCHEDULES): vol.All(
+            cv.ensure_list, [SERVICE_CHARGE_SET_SCHEDULE_SCHEMA]
+        ),
+    }
+)
+SERVICE_CHARGE_START_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_VIN): cv.matches_regex(REGEX_VIN),
+    }
+)
+
+SERVICE_AC_CANCEL = "ac_cancel"
+SERVICE_AC_START = "ac_start"
+SERVICE_CHARGE_SET_MODE = "charge_set_mode"
+SERVICE_CHARGE_SET_SCHEDULES = "charge_set_schedules"
+SERVICE_CHARGE_START = "charge_start"
+SERVICES = [
+    SERVICE_AC_CANCEL,
+    SERVICE_AC_START,
+    SERVICE_CHARGE_SET_MODE,
+    SERVICE_CHARGE_SET_SCHEDULES,
+    SERVICE_CHARGE_START,
+]
+
+
+def setup_services(hass: HomeAssistant) -> None:
+    """Register the Renault services."""
+
+    async def ac_cancel(service_call: ServiceCall) -> None:
+        """Cancel A/C."""
+        proxy = get_vehicle_proxy(service_call.data)
+
+        LOGGER.debug("A/C cancel attempt")
+        result = await proxy.vehicle.set_ac_stop()
+        LOGGER.info("A/C cancel result: %s", result)
```

Comment:
This looks like debug level.
```suggestion
        LOGGER.debug("A/C cancel result: %s", result)
```